database-enclave/enclave/src/main/java/com/r3/conclave/sample/enclave/DatabaseEnclave.java

```diff
@@ -0,0 +1,105 @@
+ */
+public class DatabaseEnclave extends Enclave {
+    /**
+     * This method gets called when client wants to communicate to enclave, and sends a message wrapped in a mail to host.
+     * Host in turn calls deliverMail method which in turn
+     * calls this method. In this method, we will deserialize the mail message, perform the computation and send the
+     * result back to the clients.
+     *
+     * @param mail        decrypted and authenticated mail body
+     * @param routingHint used by enclave to tell host whom to send the reply back to
+     */
+    @Override
+    protected void receiveMail(EnclaveMail mail, String routingHint) {
+        InputData inputData = (InputData) deserialize(mail.getBodyAsBytes());
+
+        if (inputData != null) {
+            CommandType commandType = inputData.getCommandType();
+            try {
+                Class.forName("org.h2.Driver");
+                Connection conn = DriverManager.getConnection("jdbc:h2:~/test");
+                Statement st = conn.createStatement();
+
+                User user = inputData.getUser();
+
+                if (commandType == CommandType.CREATE) {
+                    st.executeUpdate("create table if not exists users(name varchar(20),  password varchar(20))");
+                    System.out.println("Created table users if table does not exists");
+                } else if (commandType == CommandType.ADD) {
+                    PreparedStatement stmt = conn.prepareStatement("INSERT INTO users VALUES (?, ?)");
+
+                    stmt.setString(1, user.getName());
+                    stmt.setString(2, user.getPassword());
+
+                    int result = stmt.executeUpdate();
+                    System.out.println("Inserted record in the database : " + result);
+
+                    ResultSet rs = st.executeQuery("SELECT * FROM users WHERE name = '" + user.getName() + "'");
+                    System.out.println("select query output : ");
+
+                    while (rs.next()) {
+                        String name = rs.getString("name");
+                        String password = rs.getString("password");
+                        System.out.println(name);
+                        System.out.println(password);
+                    }
+                    byte[] encryptedReply = postOffice(mail.getAuthenticatedSender()).
+                            encryptMail("Added name and password to the database".getBytes());
+                    postMail(encryptedReply, routingHint);
+                } else if (commandType == CommandType.VERIFY) {
+                    //retrieve the records from the database when the user passes the VERIFY command and send it back to the client
+                    ResultSet rs = st.executeQuery("SELECT * FROM users WHERE name = '" + user.getName() + "'");
+                    System.out.println("select query output : ");
```

Comment:
I am noticing that we do the select in both cases. Maybe it would be nice to have a separate function, so to avoid duplication?

Reply:
done